pkg/observability/k8s-events.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/aws/aws-node-termination-handler/pkg/monitor/rebalancerecommendation"
+	"github.com/aws/aws-node-termination-handler/pkg/monitor/scheduledevent"
+	"github.com/aws/aws-node-termination-handler/pkg/monitor/spotitn"
+	"github.com/aws/aws-node-termination-handler/pkg/monitor/sqsevent"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+)
+
+// Kubernetes event types, reasons and messages
+const (
+	Normal                  = corev1.EventTypeNormal
+	Warning                 = corev1.EventTypeWarning
+	MonitorErrReason        = "MonitorError"
+	MonitorErrMsgFmt        = "There was a problem monitoring for events in monitor '%s'"
+	UncordonErrReason       = "UncordonError"
+	UncordonErrMsgFmt       = "There was a problem while trying to uncordon the node: %s"
+	UncordonReason          = "Uncordon"
+	UncordonMsg             = "Node successfully uncordoned"
+	PreDrainErrReason       = "PreDrainError"
+	PreDrainErrMsgFmt       = "There was a problem executing the pre-drain task: %s"
+	PreDrainReason          = "PreDrain"
+	PreDrainMsg             = "Pre-drain task successfully executed"
+	CordonErrReason         = "CordonError"
+	CordonErrMsgFmt         = "There was a problem while trying to cordon the node: %s"
+	CordonReason            = "Cordon"
+	CordonMsg               = "Node successfully cordoned"
+	CordonAndDrainErrReason = "CordonAndDrainError"
+	CordonAndDrainErrMsgFmt = "There was a problem while trying to cordon and drain the node: %s"
+	CordonAndDrainReason    = "CordonAndDrain"
+	CordonAndDrainMsg       = "Node successfully cordoned and drained"
+	PostDrainErrReason      = "PostDrainError"
+	PostDrainErrMsgFmt      = "There was a problem executing the post-drain task: %s"
+	PostDrainReason         = "PostDrain"
+	PostDrainMsg            = "Post-drain task successfully executed"
+)
+
+// Interruption event reasons
+const (
+	scheduledEventReason          = "ScheduledEvent"
+	spotITNReason                 = "SpotInterruption"
+	sqsTerminateReason            = "SQSTermination"
+	rebalanceRecommendationReason = "RebalanceRecommendation"
+	unknownReason                 = "UnknownInterruption"
+)
+
+// K8sEventRecorder wraps a Kubernetes event recorder with some extra information
+type K8sEventRecorder struct {
+	annotations map[string]string
+	enabled     bool
+	node        *corev1.Node
+	record.EventRecorder
+}
+
+// InitK8sEventRecorder creates a Kubernetes event recorder
+func InitK8sEventRecorder(enabled bool, nodeName string, nodeMetadata ec2metadata.NodeMetadata, extraAnnotationsStr string) (K8sEventRecorder, error) {
+	if !enabled {
+		return K8sEventRecorder{}, nil
+	}
+
+	// Create default annotations
+	// Worth iterating over nodeMetadata fields using reflect? (trutx)
```

Comment:
a question for our team or leftover from development?

Reply:
A question/suggestion for whoever took another pass in the future. Would make annotations creation more dynamic in case new fields were added to `nodeMetadata`.

In any case, removed it along with the rest of the in-function comments.